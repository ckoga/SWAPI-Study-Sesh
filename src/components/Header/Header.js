import React from 'react'
import './Header.scss'

const Header = ({name, quote, rank, currentPage}) => {
  const buttonView = currentPage === 'landing' ? 'hide' : 'show'

  return(
    <header className='header'>
    <h1>WELCOME <span>{name}</span></h1>
    <h2>{rank}</h2>
    <p>{quote}</p>
    <button className={buttonView}>Sign Out</button>
    </header>
  )
}

export default Header;
