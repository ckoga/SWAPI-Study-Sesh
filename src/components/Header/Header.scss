@import '../../variables';

.header {
  background-color: $header-background;
  color: $accent-color;
  border: 2px solid $accent-color;
  border-left: none;
  border-right: none;
  width: 100%;
  height: 10%;
}

h1 {
  margin: 10px;
}

.hide {
 display: none;
}
