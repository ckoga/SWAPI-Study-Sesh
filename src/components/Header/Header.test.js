import React from 'react';
import {shallow} from 'enzyme';
import Header from './Header';

describe('Card', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<Header
        name='Chase'
        quote='Obscure StarWars Quote'
        rank='Ranking: Novice'
        updateUser={jest.fn()}
      />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with empty strings passed in', () => {
    const wrapper = shallow(<Header
      name=''
      quote=''
      rank=''
      updateUser={jest.fn()}
      />);

    expect(wrapper).toMatchSnapshot();
  });

});
