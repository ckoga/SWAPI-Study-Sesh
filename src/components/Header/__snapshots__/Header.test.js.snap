// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all data passed in correctly 1`] = `
<header
  className="header"
>
  <h1>
    WELCOME 
    <span>
      Chase
    </span>
  </h1>
  <h2>
    Ranking: Novice
  </h2>
  <p>
    Obscure StarWars Quote
  </p>
  <Link
    to="/"
  >
    <button
      className="show"
      onClick={[Function]}
    >
      Sign Out
    </button>
  </Link>
</header>
`;

exports[`Card should match the snapshot with empty strings passed in 1`] = `
<header
  className="header"
>
  <h1>
    WELCOME 
    <span />
  </h1>
  <h2 />
  <p />
  <Link
    to="/"
  >
    <button
      className="hide"
      onClick={[Function]}
    >
      Sign Out
    </button>
  </Link>
</header>
`;
